src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,115 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double CAPACITY_MULTIPLY_BY = 1.5;
+    private static final String NEGATIVE_INDEX_ERROR = "Index should be natural number";
+    private static final String INDEX_OUT_OF_SIZE_ERROR = "Index out of list length";
+
+    private T[] data;
+    private int size;
+
+    public ArrayList() {
+        data = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
+    public ArrayList(int capacity) {
+        if (capacity > 0) {
+            data = (T[]) new Object[capacity];
+        } else if (capacity == 0) {
+            data = (T[]) new Object[]{};
```

Comment:
if your capacity == 0, you should create arrayList with default capacity
```suggestion
            data = (T[]) new Object[DEFAULT_CAPACITY];
```